integration/integration_test.go

```diff
@@ -1142,6 +1143,150 @@ func testSessionRecordingModes(t *testing.T, suite *integrationTestSuite) {
 	}
 }
 
+// testSyncRecordingFailures given a cluster with recording mode set to
+// sync, ensures the session is terminated in case of session recording
+// failures.
+//
+// To simulate failures, we will start an auth server, proxy and ssh server
+// separated. With the session running, we'll stop the auth server, causing any
+// sync recordings to fail. This will works for both, proxy and node recordings.
+func testSyncRecordingFailures(t *testing.T, suite *integrationTestSuite) {
+	ctx := context.Background()
+	tr := utils.NewTracer(utils.ThisFunction()).Start()
+	defer tr.Stop()
+
+	for _, test := range []struct {
+		mode string
+	}{
+		{mode: types.RecordAtNodeSync},
+		{mode: types.RecordAtProxySync},
+	} {
+		t.Run(test.mode, func(t *testing.T) {
+			t.Parallel()
+
+			recConfig, err := types.NewSessionRecordingConfigFromConfigFile(types.SessionRecordingConfigSpecV2{
+				Mode: test.mode,
```

Comment:
My editor is showing me a linter warning here:

<img width="641" alt="image" src="https://github.com/gravitational/teleport/assets/13578537/35983a93-3b39-48be-9950-11d48ebc969c">